### PR TITLE
Binderize the repo

### PR DESCRIPTION
**Pull request by taylorreiter**
This PR binderizes the repository. I opened the PR against the branch `binder`. `binder` is a branch off of the current `main`. I didn't really want to make changes to the snakefile on `main` as that was what was associated with the pub. What I'll do after this PR is merged is add the documentation for how to use the binder branch on `main`. The badge I'll include in the README will point to the `binder` branch and automatically launch from there.

This all worked for me in pangeo binder (which has more resources). There are some caveats to using the binder (your protein has to be named by its genbank accession), but I'm going to include all of that how-to in a documentation readme that's a PR against `main`.

### CODE REVIEW

File: snakemake_config.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 query_protein: 
- - P63258_ACTG_BOVIN
- - P51775_ACT_GIAIN
+ - NP_001028790.1
```
Comment (mertcelebi) [inline snakemake_config.yml:2]: are these referring to the same protein? is this the genbank accession number change you mentioned in the pr description? why is that change needed?

File: environment.yml [hunk 2]
```diff
@@ -3,5 +3,5 @@ channels:
    - bioconda
    - defaults
 dependencies:
-   - snakemake-minimal=7.16.0
-   - pandas=1.5.0
+   - snakemake-minimal=7.20.0
+   - pandas=1.3.5
```
Comment (mertcelebi) [inline environment.yml:7]: interesting that we're downgrading pandas, why?

**Review by mertcelebi — COMMENTED** [review]
two qqs before i fully review

File: environment.yml (hunk 2 above)
Comment (taylorreiter) [reply environment.yml]: it wasn't building on the binder because of a python version conflict. but this question made me realize that pandas isn't actually a dependency of the workflow anymore, so I removed it. Ran the pipeline on the binder and it worked with out pandas!

File: snakemake_config.yml (hunk 1 above)
Comment (taylorreiter) [reply snakemake_config.yml]: they refer to giardia actin and bovine actin. this is the accession change I mentioned. 

So originally, `P63258_ACTG_BOVIN` was a name that Brae gave these proteins. The first string, `P63258`, is the Uniprot accession number. `ACTG_BOVIN` describes the protein (bovine actin). 3 of the "modules" in this pipeline (alignment, function, hmm) don't care about the file name of the protein. however, for the foldseek portion, we use the alphafold Uniprot predicted structures. To download these, we need to figure out the uniprot accession. Butttt when I built the pipeline to run off of BLAST results, the results are all given as GenBank accessions. So I have some code that converts between genbank accession and uniprot, and those steps make it necessary that the input is a genbank accession.

i thought about ripping out the genbank -> uniprot and just making it so that you had to start with a uniprot accession, but genbank is much bigger than uniprot so it's kind of nice to have it start there. I _think_ the pipeline will run to completion as is if there isn't a uniprot match...but i'm not 100% sure about that.

also note I changed the accessions again -- after I open this PR, I realized `NP_001028790.1` is actually a refseq accession, so now i'm actually using the genbank one. everything in this PR does run as it's supposed to on binder.

**Comment by taylorreiter** [comment]
No not at all, I would have to open this PR against `main` instead of against `binder`. I guess it wouldn't be horrible to do that since we have the first release of the code that was run on the 50k proteins and it's all on zenodo. Do you think it would be better to do that. In effect that would:
1. Pro: Have all of the binder docs, binder files, and the generalized snakefile in one place. There wouldn't be a danger of having to maintain two code bases (the `main` and `binder` branches)
2. Con: The exact snakefile used to run the 50k blast wouldn't be as visible anymore; it would be in the release and in the git commit history

**Comment by mertcelebi** [comment]
That seems like a worthwhile trade-off, I probably would go with just `main`. And launch binder from main but maybe from a particular commit. 

For this particular repo, I don't expect many big changes in the future, so we could technically make it work with the binder branch, but it could be annoying to keep two branches in sync.

File: snakemake/snakemake_combine_all_outputs.R [hunk 3]
```diff
@@ -43,20 +29,30 @@ hmm <- read_tsv(snakemake@input[['all_hmm']]) %>%
...
+  write_tsv(all, snakemake@output[['all_outputs']])
+} else {
+  
+  # else write output skipping foldseek join
+  join1 <- full_join(features, avg_pid, by = c("protein" = "prot2"))
+  all <- full_join(join1, hmm, by = c("protein" = "query_name")) %>%
+    distinct()
+  
+  write_tsv(all, snakemake@output[['all_outputs']])
+}
```
Comment (taylorreiter) [inline snakemake/snakemake_combine_all_outputs.R:58]:
```suggestion
}

```

**Review by mertcelebi — APPROVED** [review]
